Objects/Converters/ConverterRevit/ConverterRevitShared/PartialClasses/ConvertView.Schedule.cs

```diff
@@ -257,10 +258,27 @@ Element firstElement
     }
     else if (info.field.FieldType == ScheduleFieldType.Instance)
     {
+      // All shared parameters also use this type, regardless of whether they are instance or type parameters.
+      // ref: https://www.revitapidocs.com/2024/9888db7d-00d0-4fd7-a1a9-cdd1fb5fce16.htm
       if (firstElement != null)
       {
         param = firstElement.get_Parameter(builtInParameter);
-        columnMetadata["IsReadOnly"] = param?.IsReadOnly;
+
+        // if the parameter is shared, we need to check the type parameterer too
+        Parameter typeParam = null;
+        if (firstType != null)
+        {
+          typeParam = firstType.get_Parameter(builtInParameter);
+
+          // If the parameter is readonly in the element but not in the type, is a type parameter
+          if (typeParam != null && !typeParam.IsReadOnly && param != null && param.IsReadOnly)
+          {
+            fieldType = ScheduleFieldType.ElementType.ToString();
+          }
+        }
+
+        // Check IsReadOnly for both
+        columnMetadata["IsReadOnly"] = (param?.IsReadOnly ?? false) && (typeParam?.IsReadOnly ?? false);
```

Comment:
I have some concerns about this line here. Let's say that the "firstElement" in question is a regular instance parameter (not shared). Param would have a value and typeParam would (at least sometimes) be null. In this situation, this statement would always evaluate to false, regardless of the value of "param.IsReadOnly" which doesn't seem correct to me.

What are your thoughts?

Reply:
I'm happy to merge this pr in if we can resolve this. To me it looks like we need split this line and move the type param check to the if statement on line 276. Again, please let me know what you think.

Reply:
Sorry for the delay in the response. I think you are right, but I haven't had the time to change it yet.